### DISCUSSION TITLE
Encapsulation and Child -> Parent communication?

### DISCUSSION THREAD

**Discussion by yatesco** [ivanceras/sauron#67] [Q&A] [answered]
Hi there,

I notice that most of the examples have a single top level Model with a single set of Messages. How would a reusable component (e.g. a text element) notify its parent or app about a change (e.g. the input changing)?

For example, how would the `resize_wrapper` communicate back to the parent that things have been resized: https://github.com/ivanceras/sauron/blob/master/examples/data-viewer/src/views/resize_wrapper.rs?

Thanks!

**Comment by ivanceras** [comment, marked as answer]
Hello @yatesco, 

Incidentally, I'm still working on this reusable components for the past few weeks.
However, I haven't really polish the design to make it more ergonomic for the users. I'm still torn on whether to stick with The Elm architecture or use web-components(a.k.a custom elements) that is natively supported by the browser or both. Now, that you have an inquiry about it, I just push it to the [resuable-components](https://github.com/ivanceras/sauron/tree/reusable-components) branch.

An accompanying project [sfui](https://github.com/ivanceras/sfui) is using heavily using re-usability feature to create interactive [`Components`](https://github.com/ivanceras/sauron/blob/reusable-components/crates/sauron-core/src/dom/component.rs) and `Containers` which can have its own set of behaviors that the parents component don't have to know about, but only care about plugging callbacks into the child component which the child component can then trigger. 

If you are interested, feel free to play with these, however be warned that this are highly experimental and can radically change from time to time.
